tests/framework/context/test_context.py

```diff
@@ -238,8 +239,20 @@ def test_attributes(self, tmp_path, dummy_context):
         assert isinstance(dummy_context.project_path, Path)
         assert dummy_context.project_path == tmp_path.resolve()
 
+    def test_immutble_attributes(self, dummy_context):
+        with pytest.raises(AttributeError):
+            dummy_context.catalog = 1
+        with pytest.raises(AttributeError):
+            dummy_context.params = 1
+        with pytest.raises(FrozenAttributeError):
+            dummy_context.config_loader = 1
+        with pytest.raises(FrozenAttributeError):
+            dummy_context.env = 1
+        with pytest.raises(FrozenAttributeError):
+            dummy_context.project_path = 1
```

Comment:
Should we also be testing that you can't set `package_name`, `extra_params` and `hook_manager`? Or can you set those? 🤔

Reply:
In theory, you can, it's just a Python class afterall, nothing stop you to overwrite `instance._attribute` if you really want.